src/util/canonicalizeAndValidateCommand.ts

```diff
@@ -1,23 +1,41 @@
 import canonicalizeActionName from "./canonicalizeActionName";
 import canonicalizeTargets from "./canonicalizeTargets";
 import { ActionType, PartialTarget, SelectionType } from "../typings/Types";
-import { getPrimitiveTargets } from "./getPrimitiveTargets";
+import { getPartialPrimitiveTargets } from "./getPrimitiveTargets";
+import {
+  CommandArgument,
+  CommandArgumentLatest,
+} from "../core/commandRunner/types";
 
+/**
+ * Given a command argument which comes from the client, normalize it so that it
+ * conforms to the latest version of the expected cursorless command argument.
+ *
+ * @param commandArgument The command argument to normalize
+ * @returns The normalized command argument
+ */
 export function canonicalizeAndValidateCommand(
-  inputActionName: string,
-  inputPartialTargets: PartialTarget[],
-  inputExtraArgs: any[]
-) {
+  commandArgument: CommandArgument
```

Comment:
Now that we have one argument packaged together, we can just input a command argument and return a new command argument that is latest version